src/constants/routes.ts

```diff
@@ -1,7 +1,29 @@
-export const ROUTES = {
-  home: '/',
-  signIn: '/login',
-  signUp: '/register',
-  articleDetail: '/detail/',
-  userArticlePage: '/user/',
-};
+interface Route {
+  name: string;
+  href: string;
+  nav: boolean;
+}
+
+interface Routes {
+  HOME: Route;
+  SIGNIN: Route;
+  SIGNUP: Route;
+  ARTICLEDETAIL: (slug: string) => Route;
+  USERARTICLEPAGE: (userId: string) => Route;
+}
+
+export const ROUTES: Routes = {
+  HOME: { name: 'Home', href: '/', nav: true },
+  SIGNIN: { name: 'Sign in', href: '/login', nav: true },
+  SIGNUP: { name: 'Sign up', href: '/register', nav: true },
+  ARTICLEDETAIL: (slug: string) => ({
+    name: 'Detail',
+    href: `/detail/${slug}`,
+    nav: false,
+  }),
+  USERARTICLEPAGE: (userId: string) => ({
+    name: 'User Article',
+    href: `/user/${userId}`,
+    nav: false,
+  }),
+} as const;
```

Comment:
as const를 이용해서 객체의 값들을 readonly로 만들면서 타입을 좁혀주었는데요, 변수에 명시적으로 Routes라는 타입을 대입하게 되면서 오히려 타입이 넓어졌는데요. 이 부분은 :Routes를 제거하는 방식으로 리팩토링이 진행되면 좋을 것 같아요.

추가로 아래와 같이 라우트 경로에 대해서만 상수화를 진행하고, Header 컴포넌트의 nav태그 안에 들어갈 링크들은 Header 컴포넌트에서 들고 있어도 괜찮을 것 같아요!
```suggestion
// src/constants/routes.ts
export const ROUTES = {
  HOME: '/',
  LOGIN: '/login',
  REGISTER: '/register',
  ARTICLE_DETAIL: (slug: string) => `/article/${slug}`,
} as const;

// src/components/Header.tsx
const LINKS = [
  { name: 'Home', href: ROUTES.HOME},
  { name: 'Sign in', href: ROUTES.SIGNIN},
  { name: 'Sign up', href: ROUTES.SIGNUP},
];
```